components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/utils/ExportUtils.java

```diff
@@ -0,0 +1,176 @@
+import org.wso2.carbon.apimgt.rest.api.store.v1.mappings.ApplicationKeyMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.store.v1.mappings.ApplicationMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.store.v1.models.ExportedApplication;
+import org.wso2.carbon.apimgt.rest.api.store.v1.models.ExportedSubscribedAPI;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.wso2.carbon.apimgt.impl.importexport.utils.CommonUtil.createDirectory;
+
+public class ExportUtils {
+    private static final Log log = LogFactory.getLog(ExportUtils.class);
+
+    /**
+     * Retrieve all the details of an Application by name for a given user.
+     *
+     * @param appName name of the application
+     * @return {@link Application} instance
+     * @throws APIManagementException if an error occurs while retrieving Application details
+     */
+    public static Application getApplicationDetails(String appName, String username, APIConsumer apiConsumer)
+            throws APIManagementException {
+        Application application;
+        int appId = APIUtil.getApplicationId(appName, username);
+        String groupId = apiConsumer.getGroupId(appId);
+        application = apiConsumer.getApplicationById(appId);
+        if (application != null) {
+            application.setGroupId(groupId);
+            application.setOwner(application.getSubscriber().getName());
+        }
+        return application;
+    }
+
+    /**
+     * Export a given Application to a file system as zip archive.
+     * The export root location is given by {@link @path}/exported-application.
+     *
+     * @param exportApplication Application{@link Application} to be exported
+     * @param apiConsumer       API Consumer
+     * @param withKeys          Export the Application with keys or not
+     * @return Path to the exported directory with exported artifacts
+     * @throws APIManagementException If an error occurs while exporting an application to a file system
+     */
+    public static File exportApplication(Application exportApplication, APIConsumer apiConsumer,
+            ExportFormat exportFormat, Boolean withKeys) throws APIManagementException {
+        String archivePath = null;
+        String exportApplicationBasePath;
+        String appName = exportApplication.getName();
+        String appOwner = exportApplication.getOwner();
+        try {
+            // Creates a temporary directory to store the exported application artifact
+            File exportFolder = createTempApplicationDirectory(appName, appOwner);
+            exportApplicationBasePath = exportFolder.toString();
+            archivePath = exportApplicationBasePath.concat(File.separator + appOwner + "-" + appName);
+            // Files.createDirectories(Paths.get(applicationArtifactBaseDirectoryPath));
```

Comment:
Do we need this commented code segment?

Reply:
No need of it. Removed via https://github.com/wso2/carbon-apimgt/pull/9533/commits/3a93f8d10d818c2661976f267b9c087d18e05262. Thanks for noticing.